File: obj c/obj c/ViewController2.m
//
//  ViewController2.m
//  obj c
//
//  Created by Apple on 22/12/17.
//  Copyright © 2017 TryCatch. All rights reserved.
//

#import "ViewController2.h"

@interface ViewController2 ()

@end

@implementation ViewController2

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}



- (IBAction)bcBTN:(id)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
}
- (IBAction)shopbtn:(id)sender {
    UIStoryboard *sb = [UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]];
    
    UIViewController *vc1 = [sb instantiateViewControllerWithIdentifier:@"ViewController3"];
    [self presentViewController:vc1 animated:YES completion:nil];
}
@end

File: NavigationDemo/NavigationDemo/ViewController2.swift
//
//  ViewController2.swift
//  NavigationDemo
//
//  Created by APPLE on 24/12/17.
//  Copyright © 2017 APPLE. All rights reserved.
//

import UIKit

class ViewController2: UIViewController {
var val = 1
    var timer = Timer()
    override func viewDidLoad() {
        super.viewDidLoad()
 self.navigationItem.title = "PAGE 2"
        timer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(MyMehod), userInfo: nil, repeats: true)
        // Do any additional setup after loading the view.
    }
    
    @objc func MyMehod(){
        print("MethodCalled")
     
        if val == 3
        {
            self.navigationController?.popViewController(animated: true)
        }
       val=val+1
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidDisappear(_ animated: Bool) {
        timer.invalidate()
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
